### ISSUE TITLE
Generating new devices on every deploy if used in subflow with enviromental variable

### ISSUE THREAD

**Issue by Maui2k** [csuermann/node-red-contrib-virtual-smart-home#34]
Hi @csuermann,

I just tried to use your nodes within a subflow, to put the mapping that is needed for my devices inside. I used an enviromental variable to pass the name to your nodes. Sadly I received a notification on every deploy, that new devices were dicovered, for all the devices I changed to use the sub flows.

I am quite new to node-red and subflows, but I did not found any settings that could cause that.

Kind Regards
Maui

**Comment by csuermann** [comment]
Hi @Maui2k,

Do the devices appear as duplicates in the Alexa app and / or in the vsh-connection node?

**Comment by Maui2k** [comment]
Good evening @csuermann,

the device itself is only existing once as far as I see. But on every deploy it seems that a new device is created with the same name and the old one is deleted. 

I think this, because, i get a notification from the alexa app that x new devices have been discovered after deploying. Without adding any new device to node-red. And if I added the device to a group in the alexa app, after deploying again, the device was not any more in the group.

Kind regards
Maui

**Comment by csuermann** [comment]
I was able to reproduce the issue and think I know what's going on:

When you wrap a VSH node in a subflow, Node-RED dynamically creates VSH node instances for it on every deploy. Node-RED assigns random IDs to these dynamic nodes, which are used as primary identifiers for VSH.

When you (re)deploy your flows, VSH checks which device nodes have been added and adds them to the Alexa backend, too. This also triggers the push notification. At the same time, VSH also detects which previously known device nodes have been removed from the flow and removes them from the Alexa backend, too. This explains why your nodes don't appear as duplicates.

I would advise you not to use VSH nodes in subflows.
